ops/op2/README.md

```diff
@@ -15,6 +15,26 @@ string. At this time there is no way to avoid this copy, though the `op` code
 does attempt to avoid any allocations where possible by making use of a stack
 buffer.
 
+## `async` calls
+
+Asynchronous calls are supported in two forms:
+
+```rust
+async fn op_xyz(...) -> X {}
+```
+
+and
+
+```rust
+fn op_xyz(...) -> impl Future<Output = X> {}
+```
+
+These are desugared to a function that looks like so:
+
+```rust
+fn op_xyz(promise_id: u32, ...) -> Option<X> {}
```

Comment:
I don't understand it - they get desugared into something that doesn't return Future at all?

Reply:
Correct...  it's because it returns X if the eager poll was successful, and feeds the future into pending ops otherwise. I'll clarify this text here because it's not clear.